qiskit/circuit/random/utils.py

```diff
@@ -11,15 +11,394 @@
+
+    Args:
+        interaction_graph (int): Interaction Graph
+        min_2q_gate_per_edge (int): Minimum number of times every qubit-pair must be used.
+        max_operands (int): maximum qubit operands of each gate (between 1 and 2)
+        measure (bool): if True, measure all qubits at the end
+        conditional (bool): if True, insert middle measurements and conditionals
+        reset (bool): if True, insert middle resets. ( insert_1q_oper should be true )
+        seed (int): sets random seed (optional)
+        insert_1q_oper (bool): Insert 1Q operations to the qubits which are left after
+        applying a 2Q gate on the selected qubit-pair.
+        prob_conditional (float): Probability less than 1.0, this is used to control the
+        occurence of conditionals in the circuit.
+
+    Returns:
+        QuantumCircuit: constructed circuit
+
+    Raises:
+        CircuitError: when invalid options given
+        ValueError: when the probabilities for qubit-pairs are passed and they do not add to 1.0.
+        ValueError: when some edges have probability as None.
+    """
+    # Leaving out 'free_nodes'
+    pydi_graph, pydi_graph_node_map, _, _ = interaction_graph
+    edges_probs = pydi_graph.edges()
+    num_edges = pydi_graph.num_edges()
+
+    # Just a switch for the probability weighted selection of a particular qubit-pair.
+    prob_weighted_mapping = False
+
+    # If all edge weights are none, assume the weight of each edge to be 1/N.
+    # If any of the values of the probability is None, then it would raise an error.
+    if all(edges_probs):
+        prob_weighted_mapping = True
+    elif any(edges_probs):
+        raise ValueError("The probabilities of a qubit-pair getting seleted contains `None`")
+    elif None in edges_probs:
+        prob_weighted_mapping = True
+        edges_probs = [1.0 / num_edges for _ in range(num_edges)]
+
+    # edge weights in interaction_graph must be positive
+    if prob_weighted_mapping:
+        for prob in edges_probs:
+            if prob < 0:
+                raise ValueError("The probability should be positive")
+
+    # edge weights in interaction_graph should be normalized to sum up to 1.0
+    if prob_weighted_mapping and not np.isclose(np.sum(edges_probs), 1.000, rtol=0.001):
+        raise ValueError(
+            "The sum of all the probabilities of a qubit-pair to be selected is not 1.0"
+        )
+
+    if pydi_graph_node_map is not None:
+        # The number of vertices in interaction_graph define the number of qubits
+        # in the circuit to be generated
+        qubits = list(pydi_graph_node_map.keys())
+    else:
+        n_q = pydi_graph.num_nodes()
+        qubits = [Qubit(QuantumRegister(n_q, "q"), idx) for idx in pydi_graph.nodes()]
+    num_qubits = len(qubits)
+
+    if num_qubits == 0:
+        return QuantumCircuit()
+
+    if max_operands < 1 or max_operands > 2:
+        raise CircuitError("max_operands must be between 1 and 2")
```

Comment:
This function is intended to only work on single-qubit and two-qubit gates - n-qubit gates with n>2 may be supported in the future (There is a question on how to model qubit connectivity for n-qubit gates).

Reply:
Added suggestion :)